### PR TITLE
FIX: Watched words replacement emoji render

### PR DESCRIPTION
**Pull request by keegangeorge**
This PR resolves a 🐛 Bug Report brought up on [Meta](https://meta.discourse.org/t/replace-word-feature-stopped-working-with-emojis/242216).

The fix ensures that watched words replacements that trigger an emoji short code should render an emoji successfully.

For example the replacement:

`betis` to `:poop:`

Should trigger 💩 rather than `:poop:`

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js [hunk 1]
```diff
@@ -136,6 +137,80 @@ acceptance("Admin - Watched Words", function (needs) {
...
+              },
+            },
+          ],
+        },
+      });
+    });
+  });
+
+  test("emoji renders successfully after replacement", async function (assert) {
+    await visit("/admin/customize/watched_words/action/replace");
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js:192]: I don't think this first visit to the admin UI is needed.

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js [hunk 2]
```diff
@@ -136,6 +137,80 @@ acceptance("Admin - Watched Words", function (needs) {
...
+    // await this.pauseTest();
+    await visit("/");
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js:194]: Nitpick, but the test can be shorter if we visit an existing topic and trigger a reply, we wouldn't need to set the category and title in that scenario.

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js [hunk 3]
```diff
@@ -136,6 +137,80 @@ acceptance("Admin - Watched Words", function (needs) {
...
+  needs.site({
+    watched_words_replace: {
+      "(?:\\W|^)(betis)(?=\\W|$)": {
+        replacement: ":poop:",
+        case_sensitive: false,
+      },
+    },
+  });
+  needs.pretender((server, helper) => {
+    server.get("/admin/customize/watched_words.json", () => {
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js:161]: Not sure we need this pretender either.

**Comment by pmusaraj** [comment]
Can you review this when you get a chance @nbianca, you're much more familiar with watched words than me.

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js (hunk 1 above)
Comment (keegangeorge) [reply app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js:192]: Ah yes, forgot to get rid of that. Did that while testing

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js [hunk 4]
```diff
@@ -136,6 +136,41 @@ acceptance("Admin - Watched Words", function (needs) {
     assert.strictEqual(query(".modal-body li .match").innerText, "Hello");
     assert.strictEqual(query(".modal-body li .tag").innerText, "greeting");
   });
+
+  test("emoji replacement", async function (assert) {
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js:140]: This looks like a copy of the "test modal - replace" test case, doesn't look like we need it.

**Review by pmusaraj — APPROVED** [review]

File: app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js (hunk 4 above)
Comment (keegangeorge) [reply app/assets/javascripts/discourse/tests/acceptance/admin-watched-words-test.js]: Thanks, I've removed it now 👍🏽

**Pull request merged**